### PR TITLE
Adding support for mercurial meteora vaults parsing

### PR DESCRIPTION
**Pull request by ochaloup**
This seems to be a first part of Mercurial Meteora processing.

This is about vaults. Then there is need to fix Mercurial AMM pools as it works somehow strangly.

### CODE REVIEW

**Comment by ochaloup** [comment]
@butonium this is updated PR for to be aligned with changes at
https://github.com/marinade-finance/marinade-snapshot-etl/pull/11

**Comment by ochaloup** [comment]
@butonium I updated the code to fix wrong vault to be considered as using vault and fixing issue in calculation where wrong total amount was taken. What I can see it seems reasonable correct.

**Review by butonium — APPROVED** [review]

File: filters.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply filters.json]: filters.json is part of the `.gitignore` with this PR. The file is not part of the PR either.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply .vscode/settings.json]: the file is not part of the commit but I haven't added it to `.gitignore` as it's defined differently there
https://github.com/marinade-finance/solana-snapshot-manager/blob/master/.gitignore#L33

**Pull request merged**